diff --git a/src/libide/lsp/meson.build b/src/libide/lsp/meson.build
index af63abff0a321774808279e4ab7afb17216a4e1e..cdd6993c13e22ee9a011c960c2069f814bf9bdbf 100644
--- a/src/libide/lsp/meson.build
+++ b/src/libide/lsp/meson.build
@@ -92,7 +92,7 @@ libide_lsp_deps = [
 # Library Definitions
 #
 
-libide_lsp = static_library('ide-lsp-' + libide_api_version, libide_lsp_sources,
+libide_lsp = static_library('ide-lsp-' + libide_api_version, libide_lsp_sources, libide_lsp_generated_headers,
    dependencies: libide_lsp_deps,
          c_args: libide_args + release_args + ['-DIDE_LSP_COMPILATION'],
 )
