diff -Nur wpa_supplicant-2.0.orig/wpa_supplicant/wpa_supplicant.c wpa_supplicant-2.0/wpa_supplicant/wpa_supplicant.c
--- wpa_supplicant-2.0.orig/wpa_supplicant/wpa_supplicant.c	2013-01-12 09:42:53.000000000 -0600
+++ wpa_supplicant-2.0/wpa_supplicant/wpa_supplicant.c	2013-05-11 14:09:34.586718122 -0500
@@ -1666,10 +1666,10 @@
 
 		if (assoc_failed) {
 			/* give IBSS a bit more time */
-			timeout = ssid->mode == WPAS_MODE_IBSS ? 10 : 5;
+			timeout = ssid->mode == WPAS_MODE_IBSS ? 20 : 10;
 		} else if (wpa_s->conf->ap_scan == 1) {
 			/* give IBSS a bit more time */
-			timeout = ssid->mode == WPAS_MODE_IBSS ? 20 : 10;
+			timeout = ssid->mode == WPAS_MODE_IBSS ? 20 : 20;
 		}
 		wpa_supplicant_req_auth_timeout(wpa_s, timeout, 0);
 	}
