--- ./configure.ac.orig	2015-04-09 06:28:58.000000000 -0500
+++ ./configure.ac	2015-04-16 15:33:30.087481982 -0500
@@ -114,19 +114,19 @@
     fi
 fi
 
-AX_CHECK_PYTHON_MAKO_MODULE($PYTHON_MAKO_REQUIRED)
+dnl AX_CHECK_PYTHON_MAKO_MODULE($PYTHON_MAKO_REQUIRED)
 
-if test -z "$PYTHON2"; then
-    if test ! -f "$srcdir/src/util/format_srgb.c"; then
-        AC_MSG_ERROR([Python not found - unable to generate sources])
-    fi
-else
-    if test "x$acv_mako_found" = xno; then
-        if test ! -f "$srcdir/src/mesa/main/format_unpack.c"; then
-            AC_MSG_ERROR([Python mako module v$PYTHON_MAKO_REQUIRED or higher not found])
-        fi
-    fi
-fi
+dnl if test -z "$PYTHON2"; then
+dnl     if test ! -f "$srcdir/src/util/format_srgb.c"; then
+dnl         AC_MSG_ERROR([Python not found - unable to generate sources])
+dnl     fi
+dnl else
+dnl     if test "x$acv_mako_found" = xno; then
+dnl         if test ! -f "$srcdir/src/mesa/main/format_unpack.c"; then
+dnl             AC_MSG_ERROR([Python mako module v$PYTHON_MAKO_REQUIRED or higher not found])
+dnl         fi
+dnl     fi
+dnl fi
 
 AC_PROG_INSTALL
 
