--- a/avahi-core/server.c
+++ b/avahi-core/server.c
@@ -1582,9 +1582,9 @@ AvahiServerConfig* avahi_server_config_init(AvahiServerConfig *c) {
     c->host_name = NULL;
     c->domain_name = NULL;
     c->check_response_ttl = 0;
-    c->publish_hinfo = 1;
+    c->publish_hinfo = 0;
     c->publish_addresses = 1;
-    c->publish_workstation = 1;
+    c->publish_workstation = 0;
     c->publish_domain = 1;
     c->use_iff_running = 0;
     c->enable_reflector = 0;

--- a/man/avahi-daemon.conf.5.xml.in
+++ b/man/avahi-daemon.conf.5.xml.in
@@ -244,7 +244,7 @@
       useful for administrative purposes. This is recommended by the
       mDNS specification but not required. For the sake of privacy
       you might choose to disable this feature. Defaults to
-      "yes."</p>
+      "no".</p>
     </option>
 
     <option>
@@ -254,7 +254,7 @@
       might be useful for administrative purposes (i.e. browse for
       all PCs on the LAN), but is not required or recommended by any
       specification. Newer MacOS X releases register a service of
-      this type. Defaults to "yes".</p>
+      this type. Defaults to "no".</p>
     </option>
 
     <option>
--- a/avahi-daemon/avahi-daemon.conf
+++ b/avahi-daemon/avahi-daemon.conf
@@ -21,7 +21,7 @@
 [server]
 #host-name=foo
 #domain-name=local
-browse-domains=0pointer.de, zeroconf.org
+#browse-domains=0pointer.de, zeroconf.org
 use-ipv4=yes
 use-ipv6=no
 #allow-interfaces=eth0
@@ -46,8 +46,8 @@ enable-wide-area=yes
 #disable-user-service-publishing=no
 #add-service-cookie=no
 #publish-addresses=yes
-#publish-hinfo=yes
-#publish-workstation=yes
+publish-hinfo=no
+publish-workstation=no
 #publish-domain=yes
 #publish-dns-servers=192.168.50.1, 192.168.50.2
 #publish-resolv-conf-dns-servers=yes
