diff --git a/libgitg/gitg-diff-view-file-selectable.vala b/libgitg/gitg-diff-view-file-selectable.vala
index 369d7ae644c81dad885ac35092aa87e9bc5a31d2..46673c6fa8f26917bc436244999e5379ca1e9793 100644
--- a/libgitg/gitg-diff-view-file-selectable.vala
+++ b/libgitg/gitg-diff-view-file-selectable.vala
@@ -54,7 +54,7 @@ class Gitg.DiffViewFileSelectable : Object
 
 		buffer.get_start_iter(out iter);
 
-		while (buffer.forward_iter_to_source_mark(ref iter, d_selection_category))
+		while (buffer.forward_iter_to_source_mark(iter, d_selection_category))
 		{
 			ret += iter.get_line();
 		}
@@ -321,7 +321,7 @@ class Gitg.DiffViewFileSelectable : Object
 
 		var buffer = source_view.buffer as Gtk.SourceBuffer;
 
-		if (!buffer.forward_iter_to_source_mark(ref iter, "header"))
+		if (!buffer.forward_iter_to_source_mark(iter, "header"))
 		{
 			iter.forward_to_end();
 		}
@@ -493,7 +493,7 @@ class Gitg.DiffViewFileSelectable : Object
 		}
 		else
 		{
-			something_selected = (buffer as Gtk.SourceBuffer).forward_iter_to_source_mark(ref iter, d_selection_category);
+			something_selected = (buffer as Gtk.SourceBuffer).forward_iter_to_source_mark(iter, d_selection_category);
 		}
 
 		if (something_selected != has_selection)
diff --git a/meson.build b/meson.build
index fada8986da96d387bb5d2a17fdbd5affbbdb5df5..7b27b8ece6f72edab6ed797e49bf445d229160fd 100644
--- a/meson.build
+++ b/meson.build
@@ -132,7 +132,7 @@ gobject_introspection_dep = dependency('gobject-introspection-1.0', version: '>=
 gsettings_desktop_schemas_dep = dependency('gsettings-desktop-schemas')
 gtk_dep = dependency('gtk+-3.0', version: '>= 3.20.0')
 gtkspell_dep = dependency('gtkspell3-3.0')
-gtksourceview_dep = dependency('gtksourceview-3.0', version: '>= 3.10')
+gtksourceview_dep = dependency('gtksourceview-4', version: '>= 4.0.3')
 libgit2_glib_dep = dependency('libgit2-glib-1.0', version: ['>= 0.27.8'])
 libpeas_dep = dependency('libpeas-1.0')
 libsecret_dep = dependency('libsecret-1')
diff --git a/org.gnome.gitgDevel.json b/org.gnome.gitgDevel.json
index 4349a709d98e3f64b086fc9ac77d86f901157d4d..08acddb2602065b4f4f514d8fc5b49c686e0b480 100644
--- a/org.gnome.gitgDevel.json
+++ b/org.gnome.gitgDevel.json
@@ -140,7 +140,7 @@
                 {
                     "type" : "git",
                     "url" : "https://gitlab.gnome.org/GNOME/gtksourceview.git",
-                    "tag" : "gnome-3-24"
+                    "tag" : "gtksourceview-4-2"
                 }
             ]
         },
