--- configure.ac.orig	2014-10-24 20:22:10.538904539 +0200
+++ configure.ac	2014-10-24 20:23:31.334904994 +0200
@@ -243,7 +243,7 @@
               [enable_systemd=auto])
 
 if test x$enable_systemd != xno; then
-   PKG_CHECK_MODULES(SYSTEMD, [libsystemd-login >= 186],
+   PKG_CHECK_MODULES(SYSTEMD, [libsystemd >= 209],
 			                     [have_systemd=yes], [have_systemd=no])
 else
    have_systemd=no
